### ISSUE TITLE
different results for innerJoin

### ISSUE THREAD

**Issue by SvenSeemann** [oguimbal/pg-mem#84]
I tried using pg-mem for my lib, which uses knex.js as an adapter.
When running my test suite I noticed there is given a different result for innerJoins when using pg-mem and using pg with a local running postgres instance.

My test case is like 

```
TableGroups
-------------
id
name


TableUsersGroup
-----------------
group_id
user_id

===>

knex('TableGroups')
    .innerJoin('TableUsersGroup', 'group_id', 'id')
    .where('user_id', 'SomeTestUserId')
```

When running my code with pg/postgres it returns
```
{
  "id": "71e65e4a-f3b6-497b-a883-036fc3f313e5",
  "name": "TestGroup",
  "user_id": "3c7c772e-bb92-414e-8c83-44f50fbf43ec",
  "group_id": "71e65e4a-f3b6-497b-a883-036fc3f313e5"
}
```

When running my code with pg-mem it returns
```
{
  ">joined": {
    "group_id": "acfad3e9-f60f-4ec2-9a92-300026da9309",
    "user_id": "3c7c772e-bb92-414e-8c83-44f50fbf43ec"
  },
  ">restrictive": {
    "id": "acfad3e9-f60f-4ec2-9a92-300026da9309",
    "name": "TestGroup"
  }
}
```

**Comment by oguimbal** [comment]
Oh, yes... that's an implementation detail that is leaking out of pg-mem.
I think this bug is actually solved in the master branch, but it is not ready to be published (unfortunately I dont have time these days, but I'ill try to have a look this weekend !)

Thanks for the report.

**Comment by oguimbal** [comment]
Hi ! This should be fixed with pg-mem@1.8.4

Please reopen this issue if thats not the case :)
